src/gourd/post/afterscript.rs

```diff
@@ -1,113 +1,70 @@
-use std::fs;
-use std::os::unix::fs::PermissionsExt;
-use std::path::Path;
-use std::path::PathBuf;
-use std::process::ExitStatus;
-
 use anyhow::anyhow;
-use anyhow::Context;
 use anyhow::Result;
-use gourd_lib::bailc;
-use gourd_lib::ctx;
 use gourd_lib::experiment::Experiment;
+use gourd_lib::file_system::FileOperations;
 use gourd_lib::resources::run_script;
 use log::debug;
 use log::trace;
 
-/// Runs the afterscript on jobs that are completed and do not yet have an
-/// afterscript output.
-pub fn run_afterscript(run_id: usize, experiment: &Experiment) -> Result<()> {
+/// For a run that:
+/// * has finished
+/// * its program has an afterscript
+///
+/// this function will run said afterscript, and update the experiment
+/// accordingly.
+pub fn run_afterscript(run_id: usize, experiment: &mut Experiment) -> Result<()> {
     let run = &experiment.runs[run_id];
-    let after_out_path = &run.afterscript_output_path;
-    let res_path = run.output_path.clone();
+    let run_output_path = run.output_path.clone();
 
     trace!("Checking afterscript for {run_id}");
 
-    let after_output = after_out_path
-        .clone()
-        .ok_or(anyhow!("Could not get the afterscript information"))
-        .with_context(ctx!(
-            "Could not get the afterscript information", ;
-            "",
-        ))?;
-
     let afterscript = &experiment.programs[run.program]
         .afterscript
         .clone()
-        .ok_or(anyhow!("Could not get the afterscript information"))
-        .with_context(ctx!(
-            "Could not get the afterscript information", ;
-            "",
-        ))?;
+        .ok_or(anyhow!("Could not get the afterscript information"))?;
 
     debug!("Running afterscript for {run_id}");
-    let exit_status =
-        run_afterscript_for_run(afterscript, &res_path, &after_output, &run.work_dir)?;
+    let afterscript_output = run_script(
+        &afterscript.executable,
+        vec![&run_output_path.display().to_string()],
+        &run.work_dir,
+    )?;
 
-    if !exit_status.success() {
-        bailc!("Afterscript failed with exit code {}",
-                exit_status
-                    .code()
-                    .ok_or(anyhow!("Status does not exist"))
-                    .with_context(ctx!(
-                        "Could not get the exit code of the execution", ;
-                        "",
-                    ))? ; "", ; "", );
-    }
+    let afterscript_result = String::from_utf8_lossy(&afterscript_output.stdout)
+        .trim()
+        .to_string();
+    debug!("stdout: {afterscript_result}");
```

Comment:
This debug statement is unhelpful.

Try: `afterscript: ..., said: ...`

Reply:
agreed! will fix

Reply:
Thanks!